Deposit all data to Zenodo

BharatVe opened the pull request:
Closes #63

Updates an existing draft only (requires --deposition-id or a setting).
Never publishes (manual step stays in Zenodo UI).
Uploads: README.md, optimap-main.zip, latest GeoJSON (.gz) and GeoPackage (.gpkg) (reuses existing dump tasks/paths).
Merges metadata non-destructively (no clobber):
Reads data/zenodo_dynamic.json and merges: version, related_identifiers (live dump links + “describes” per source).
Does NOT touch stable fields: creators, grants/funders, language, contributors, communities, upload_type, DOI, publisher.
Robust error handling around /actions/edit, 4xx/5xx, and bucket uploads.

How it works -Admin runs:

```
python manage.py render_zenodo_desc
python manage.py deploy_zenodo --confirm --deposition-id <draft_id>
```
First command creates/updates:

- data/optimap-main.zip
- data/README.md
- data/zenodo_dynamic.json

Second command:

- Reads those files, merges dynamic metadata, uploads files to the existing sandbox draft.
- Leaves stable metadata untouched and does not publish.

BharatVe committed:
Upload to zenodo with tests

BharatVe committed:
Update requirements-dev.txt

nuest commented on tests/test_end_to_end_zenodo.py:
```diff
@@ -0,0 +1,196 @@
+    def tearDown(self):
+        self._tmpdir.cleanup()
+
+    def test_end_to_end_render_then_deploy(self):
```
This test case is too big. You should be able to split it up at least into 

1. file generation
2. deposition

You're welcome to copy and paste the file generation code from the first test to the second.

nuest commented on tests/test_end_to_end_zenodo.py:
```diff
@@ -0,0 +1,196 @@
+        (self.data_dir / "optimap_data_dump_20250101.gpkg").write_bytes(b"GPKG")
+
+        # ---- Mock Zenodo API + uploader
+        existing = {
```
For readability of the actual test, you can put these objects at the top of the file as immutable objects.

nuest reviewed (changes requested):
Very good first draft for the PR - thank you!

BharatVe committed:
Ignore generated ZIP artifact

BharatVe committed:
Ignore generated ZIP artifact

BharatVe committed:
Updated tests and files, fixed changes requested in comments

BharatVe committed:
Merge remote-tracking branch 'origin/enhancement/Create_new_record_on_Zenodo_63' into backup/enhancement/Create_new_record_on_Zenodo_63-before-pull

BharatVe commented:
Implemented all requested changes except the two called out below. Also moved markdown into requirements.txt, removed the committed ZIP from history, renamed the command to deposit_zenodo, added logging, validated API base (no trailing /), centralized cache path via settings, and split the large test into render/deposit tests.

Still pending from the review:

TWe currently validate the Zenodo token at command runtime (inside deposit_zenodo) rather than failing in settings.py at import time.

We log a concise diff (changed keys + counts); a full pretty-printed JSON diff is not implemented.

BharatVe committed:
Minor fixes for CI tests

BharatVe committed:
Update requirements.txt
